src/main/resources/META-INF/core.xml

```diff
@@ -896,88 +896,119 @@
-               text="Show Macro Expansion"
-               class="org.rust.ide.actions.macroExpansion.RsShowMacroExpansionGroup">
-            <add-to-group group-id="EditorPopupMenu"/>
-            <separator/>
-            <reference id="Rust.ShowRecursiveMacroExpansionAction"/>
-            <reference id="Rust.ShowSingleStepMacroExpansionAction"/>
-        </group>
+        <action id="Cargo.RunCargoCommand"
+                class="org.rust.cargo.runconfig.command.RunCargoCommandAction"
+                text="Run Cargo Command"/>
 
-        <action class="org.rust.ide.refactoring.generateConstructor.GenerateConstructorAction"
-                id="Rust.GenerateConstructor"
-                text="Generate Constructor">
-            <add-to-group group-id="GenerateGroup"/>
+        <action id="Cargo.RunExternalLinter"
+                class="org.rust.ide.actions.RsRunExternalLinterAction"
+                text="Run External Linter"/>
+
+        <action id="Cargo.ToggleExternalLinterOnTheFlyAction"
+                class="org.rust.ide.actions.ToggleExternalLinterOnTheFlyAction"
+                text="Run External Linter (Cargo Check / Clippy) on the Fly"/>
+
+        <action id="Cargo.ExpandAll" class="org.rust.ide.actions.CargoTreeAction$ExpandAll" text="Expand All"
+                icon="AllIcons.Actions.Expandall"
+                use-shortcut-of="ExpandAll"/>
+
+        <action id="Cargo.CollapseAll" class="org.rust.ide.actions.CargoTreeAction$CollapseAll" text="Collapse All"
+                icon="AllIcons.Actions.Collapseall"
+                use-shortcut-of="CollapseAll"/>
+
+        <action id="Cargo.ToggleOfflineAction"
+                class="org.rust.ide.actions.ToggleOfflineAction"
+                text="Toggle Offline Mode"
+                description="Toggle offline mode for Cargo"
+                icon="AllIcons.Actions.OfflineMode"/>
+
+        <action id="Cargo.ShowSettings"
+                class="org.rust.ide.actions.CargoEditSettingsAction" text="Cargo Settings"
+                description="Edit Cargo settings for the current project"
+                icon="AllIcons.General.Settings">
         </action>
+
         <group id="Rust.Cargo">
-            <reference id="Rust.RefreshCargoProject"/>
-            <reference id="Rust.CargoCommand"/>
-            <reference id="Rust.RustfmtCargoProject"/>
-            <reference id="Rust.RunExternalLinter"/>
+            <reference id="Cargo.RefreshCargoProject"/>
+            <reference id="Cargo.AttachCargoProject"/>
+            <reference id="Cargo.DetachCargoProject"/>
+            <separator/>
+            <reference id="Cargo.RunCargoCommand"/>
+            <reference id="Cargo.RunExternalLinter"/>
+            <reference id="Cargo.RustfmtCargoProject"/>
+            <separator/>
+            <reference id="Cargo.ExpandAll"/>
+            <reference id="Cargo.CollapseAll"/>
+            <separator/>
+            <reference id="Cargo.ToggleOfflineAction"/>
```

Comment:
Are we sure that we need this action here?
It looks quite rare action for tool window.
Also, at this moment icon is rather confising

Reply:
Fixed.